beacon_node/lighthouse_network/src/discovery/mod.rs

```diff
@@ -201,8 +201,24 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
         info!(log, "ENR Initialised"; "enr" => local_enr.to_base64(), "seq" => local_enr.seq(), "id"=> %local_enr.node_id(),
               "ip4" => ?local_enr.ip4(), "udp4"=> ?local_enr.udp4(), "tcp4" => ?local_enr.tcp6()
         );
-
-        let listen_socket = SocketAddr::new(config.listen_address, config.discovery_port);
+        let mut discv5_config = config.discv5_config.clone();
```

Comment:
You clone it and mutate it, but on line 226 we still pass the old one in. 

Need to pass `discv5_config` to `Discv5::new()`

Reply:
this whole code was wrong